dotCMS/src/main/java/com/dotmarketing/business/LayoutAPIImpl.java

```diff
@@ -171,14 +169,40 @@ public int compare(Layout l1, Layout l2) {
 		return layouts;
 	}
 
+	/* this method is used to check if the user has access to edit the page portlet
+	* all the users should have access to Edit Page, regardless of the assigned portlets.
+	*/
+	private boolean editPagePortletAccess(final User user) throws DotDataException {
+		final RoleAPI roleAPI = APILocator.getRoleAPI();
+		//verify the roles of the user
+		final List<Role> foundRoles = roleAPI.loadRolesForUser( user.getUserId(), false );
+		final PermissionAPI permAPI = APILocator.getPermissionAPI();
+
+		for (final Role role: foundRoles){
+			//get the permissions for the role
+			final List<Permission> perms = permAPI.getPermissionsByRole(role, true, true);
+			//determine if the user can edit the page
+			for(final Permission perm : perms){
+				if((perm.getType().equals(IHTMLPage.class.getCanonicalName()) || perm.getType().equals(Contentlet.class.getCanonicalName()))
+						&& (perm.getPermission() == PermissionAPI.PERMISSION_EDIT || perm.getPermission() == 3)){
```

Comment:
isn't there a constant for that number 3?

Reply:
can it be replaced by a more meaning ful const? Also That validation could be a separate functon

Reply:
what is permissions 3 we have those ones:
```
final int PERMISSION_READ = 1;
	final int PERMISSION_USE = 1;
	final int PERMISSION_EDIT = 2;
	final int PERMISSION_WRITE = 2;
	final int PERMISSION_PUBLISH = 4;
	final int PERMISSION_EDIT_PERMISSIONS = 8;
	final int PERMISSION_CAN_ADD_CHILDREN = 16;
	```

Reply:
> what is permissions 3 we have those ones:
> 
> ```
> final int PERMISSION_READ = 1;
> 	final int PERMISSION_USE = 1;
> 	final int PERMISSION_EDIT = 2;
> 	final int PERMISSION_WRITE = 2;
> 	final int PERMISSION_PUBLISH = 4;
> 	final int PERMISSION_EDIT_PERMISSIONS = 8;
> 	final int PERMISSION_CAN_ADD_CHILDREN = 16;
> 	```
> ```

Permission 3 is when the user is capable of edit and view the content

Reply:
I made a refactor for the validations following your suggestions